contrib/docker/Dockerfile

```diff
@@ -0,0 +1,78 @@
+FROM alpine:3.9
+
+MAINTAINER Andrew V. Jones "andrew.jones@vector.com"
+
+ENV LC_ALL en_US.UTF-8
+ENV LANG en_US.UTF-8
+
+#
+# Deps needed only for build
+#
+ENV BUILD_DEPS \
+    alpine-sdk \
+    cmake \
+    cython \
+    cython-dev \
+    g++ \
+    gcc \
+    git \
+    make \
+    python-dev \
+    sed
+
+#
+# Deps needed for running afterwards
+#
+ENV PERSISTENT_DEPS \
+    bash \
+    libgcc \
+    libstdc++ \
+    python
+
+#
+# Steps to build-up our image
+#
+RUN apk upgrade --update && \
+    apk add --no-cache --virtual .build-deps $BUILD_DEPS && \
+    apk add --no-cache --virtual .persistent-deps $PERSISTENT_DEPS && \
+    # \
+    # Musl does not provide /usr/include/sys/unistd.h, which Lingeling needs \
+    # \
+    echo "#include <unistd.h>" > /usr/include/sys/unistd.h && \
+    git clone https://github.com/Boolector/boolector.git && \
+    cd boolector && \
+    # \
+    # Musl does not provide quite enough of glibc, so we pretend we're Windows \
+    # \
+    sed -i "s#MINGW32#Linux#g" ./contrib/setup-btor2tools.sh && \
+    sed -i "s#MINGW32#Linux#g" ./contrib/setup-lingeling.sh && \
+    sed -i "s#MINGW32#Linux#g" ./contrib/setup-picosat.sh && \
+    sed -i "s#MINGW32#Linux#g" ./contrib/setup-cadical.sh && \
+    ./contrib/setup-btor2tools.sh && \
+    ./contrib/setup-picosat.sh && \
+    # \
+    # PicoSAT has has a race-condition in its makefile, so we do a "double make" to avoid it \
+    # \
+    make -C deps/picosat -j1 && \
+    ./contrib/setup-cadical.sh && \
+    ./contrib/setup-lingeling.sh && \
+    ./configure.sh --python --shared && \
+    cd build && \
+    make -j12 && \
```

Comment:
Would be good to use a more generic approach (if `nproc` is available in Alpine):

```suggestion
    make -j$(nproc) && \
```

Reply:
Agreed! That's a copy-paste from my host build!

Reply:
Unresolving until I actually do this (and check it works).

Reply:
Done.